src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySink.java

```diff
@@ -92,6 +107,18 @@ protected BigQuerySinkConfig getConfig() {
   protected void prepareRunValidation(BatchSinkContext context) {
     FailureCollector collector = context.getFailureCollector();
     config.validate(context.getInputSchema(), config.getSchema(collector), collector);
+    try {
+      config.validatePartitionProperties(context.getInputSchema(), config.getSchema(collector), collector);
+    } catch (InvalidConfigPropertyException e) {
```

Comment:
i'm confused by this, don't see where validatePartitionProperties throws an exception

Reply:
It makes calls to helper methods which eventually propagates down to BigQueryUtils.getBigQueryTable which throws these (InvalidConfigPropertyException, InvalidStageException) as runtime exceptions.